stream.go

```diff
@@ -1013,7 +1013,8 @@ func (cs *clientStream) CloseSend() error {
 			binlog.Log(cs.ctx, chc)
 		}
 	}
-	// We never returned an error here for reasons.
+	// We don't return an error here as we expect users to read all messages
+	// from the stream and get the RPC status from RecvMsg().
```

Comment:
How about being a little more explicit why it's OK to not return an error here, but we must when doing `SendMsg`:

```suggestion
	// We don't return an error here as we expect users to read all messages
	// from the stream and get the RPC status from RecvMsg().  Note that
	// SendMsg() must return an error when one occurs so the application
	// knows to stop sending messages, but that does not apply here.
```

Reply:
Accepted the suggestion.